cmd/restic/cmd_rewrite.go

```diff
@@ -152,14 +157,46 @@ func rewriteSnapshot(ctx context.Context, repo *repository.Repository, sn *resti
 				return restic.ID{}, err
 			}
 			ss := querySize()
-			if sn.Summary != nil {
-				sn.Summary.TotalFilesProcessed = ss.FileCount
-				sn.Summary.TotalBytesProcessed = ss.FileSize
+			if sn.Summary == nil { // change of logic: create summary if it wasn't there before
+				sn.Summary = &restic.SnapshotSummary{}
 			}
+			sn.Summary.TotalFilesProcessed = ss.FileCount
+			sn.Summary.TotalBytesProcessed = ss.FileSize
 			return id, err
 		}
 
+	} else if opts.SnapshotSummary {
+		if sn.Summary != nil {
+			Printf("snapshot %s has already got snapshot summary data\n", sn.ID().Str())
+			return false, nil
+		}
+
+		rewriteNode := func(node *restic.Node, path string) *restic.Node {
+			return node // always include everything
+		}
+
+		rewriter, querySize := walker.NewSnapshotSizeRewriter(rewriteNode)
+
+		filter = func(ctx context.Context, sn *restic.Snapshot) (restic.ID, error) {
+			id, err := rewriter.RewriteTree(ctx, repo, "/", *sn.Tree)
+			if err != nil {
+				return restic.ID{}, err
+			}
+			ss := querySize()
+			if sn.Summary == nil {
+				sn.Summary = &restic.SnapshotSummary{}
+			}
+			sn.Summary.TotalFilesProcessed = ss.FileCount
+			sn.Summary.TotalBytesProcessed = ss.FileSize
+			Verbosef("totalFilesProcessed %12d\n", ss.FileCount)
+			Verbosef("totalBytesProcessed %12d\n", ss.FileSize)
+
+			return id, nil
+		}
+
 	} else {
+		// TODO: question: should metadata modification be changed so that
```

Comment:
Metadata modifications are expected to be fast operations, so I think we should keep the current behavior.